scripts/provision-group-check-users.js

```diff
@@ -0,0 +1,324 @@
+#!/usr/bin/env node
+import { ArgumentParser } from 'argparse';
+import {
+    CognitoIdentityProviderClient,
+    ListUsersCommand,
+    AdminListGroupsForUserCommand
+  } from '@aws-sdk/client-cognito-identity-provider';
+import fs from 'fs';
+import yaml from 'js-yaml';
+import process from 'process';
+import { Group } from '../src/groups.js';
+import { reportUnhandledRejectionsAtExit } from '../src/utils/scripts.js';
+/**
+ * COGNITO_USER_POOL_ID will read from the eponymous env variable or look in a
+ * config file. The config file is set via the env variable `CONFIG_FILE`, or
+ * either env/production/config.json or env/testing/config.json depending on
+ * your environment (NODE_ENV) */
+import { COGNITO_USER_POOL_ID, PRODUCTION } from '../src/config.js';
+
+const DESCRIPTION = `
+  A helper script to check provided emails against existing cognito users
+  in the pool. The CLI interface is the same as 'provision-group' however
+  no changes will be made (to cognito) by this script.
+
+  We attempt to check various scenarios (email matches, username matches) and print
+  verbose output so that we can update existing users where possible.
+
+  P.S. Set 'CONFIG_FILE=env/production/config.json' to use the production cognito
+  user pool; by default we'll use the pool defined in 'env/testing/config.json'
+  or the env variable 'COGNITO_USER_POOL_ID' if set.
+`;
+
+const REGION = COGNITO_USER_POOL_ID.split("_")[0];
+const cognito = new CognitoIdentityProviderClient({ region: REGION });
+
+function parseArgs() {
+    const argparser = new ArgumentParser({description: DESCRIPTION});
+    argparser.addArgument("groupName", {metavar: "<name>", help: "Name of the Nextstrain Group"});
+    argparser.addArgument("--members", {
+      dest: "membersFile",
+      metavar: "<file.yaml>",
+      required: true,
+      help: `
+        A YAML file describing the members to add to the Group.
+        The file must be an array of objects (i.e.  dict/map/hash).
+        Each object must contain an "email" key which is case-sensitive.
+        The "username" key is optional for this script, but is required when you actually create provision the group membership.
```

Comment:
Why allow "username" to be optional in this script? I think it should be required to match the other script, and also simplify the code by avoiding the need to handle its absence.